odl/trafos/wavelet.py

```diff
@@ -750,11 +828,26 @@ def __init__(self, ran, nscales, wbasis, mode):
 
             'per': periodization -- like periodic-padding but gives the
             smallest possible number of decomposition coefficients.
+
+            With J.-O. Stomberg's biorthogonal wavelets only signal
+            extension mode 'sym' is available.
         """
         self.nscales = int(nscales)
-        self.wbasis = wbasis
         self.mode = str(mode).lower()
 
+        if isinstance(wbasis, pywt.Wavelet):
+            self.wbasis = wbasis
+        else:
+            try:
+                wbasis = wbasis + ''
+                if wbasis.startswith('JOS'):
+                    self.wbasis = wbasis
```

Comment:
Right, we could actually just clone that pattern with a `backends` directory and push the `size_list` and stuff to the `pywavelets` part and `jos` to  another file there. And then we would just need three operators here, `WaveletTransform`, `WaveletTransformAdjoint` and `WaveletTransformInverse`. Regarding names, we have somehow settled for this pattern since it will put the names together in auto-completion lists. Sorry @niinimaki, I know I said differently a while ago. I do the restructuring.